packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1126,9 +1126,17 @@ class RouteObserver<T extends Route<dynamic>> extends NavigatorObserver {
   void subscribe(RouteAware routeAware, T route) {
     assert(routeAware != null);
     assert(route != null);
-    if (!_listeners.containsKey(route)) {
+
+    Set<RouteAware> subscribers = _listeners[route];
+
+    if (subscribers == null) {
+      subscribers = new Set<RouteAware>();
+
+      _listeners[route] = subscribers;
+    }
+
+    if (subscribers.add(routeAware)) {
       routeAware.didPush();
```

Comment:
i wonder why we call didPush here. That seems weird.

Reply:
True, it was weird. Just kept not to break something else.

Reply:
I'll look into it separately, let's leave it as-is for this patch.